analysis/mailchimp/mailchimp_gains_losses.sql

```diff
@@ -0,0 +1,47 @@
+with
```

Comment:
the performance of this query is very poor because you're repeatedly going to `mailchimp_email_summary` to get datasets that you don't need to get from that table. you use a total of three queries to that table, and each of those queries hits all of the underlying tables. in reality, each of those three queries could hit a smaller dataset. for example, `losses` could query `bounces` and `unsubscribes`.

the current version took 3:27 to run on our 8-node cluster on a test dataset :) 

I think the summary table is very useful for when we want to write simple queries that just hit it once and do aggregations on it. But I don't think it should be used as a building block like this because of the inefficiency.